### PR TITLE
Adds jekyll-assets to pipeline to minify and cachebust

### PR DESCRIPTION
**Pull request by rchild-okta**
## Description:
This PR is split up into 5 commits:

1. **Adds jekyll-assets to pipeline to minify and cachebust**

2. **Converts javascript files**

    - Moves from source -> _source
    - Adds jquery to vendor folder instead of loading from external cdn
    - Uses jekyll-assets to bundle

3. **Converts font files**

    - Moves from source -> _source
    - Replaces url with jekyl_assets font_url

4. **Converts img files**

    - Moves from source -> _source
    - Consolidates image and img to just img
    - Replaces url with image_url in sass files
    - Replaces image references in md and html files with {% img src %}

    *Minor:*
    - Removes extra AnnouncementBar.scss
    - Frontmatter is no longer necessary in main sass files
    - Cleans up some strange formatting
    - Adds back some missing page-* files from root
    - Removes inline bg image styles

5. **Adds htmlproofer and fixes broken links**

## Type of Pull Request:
- [x] Functional (CSS changes, JS updates, or layout modifications)

### Resolves:
* [OKTA-125319](https://oktainc.atlassian.net/browse/OKTA-125319)
* [OKTA-125506](https://oktainc.atlassian.net/browse/OKTA-125506)

## How Has This Been Tested?
- [x] I have built this locally and verified that it does not break existing functionality.
- [x] Ran [htmlproofer](https://github.com/gjtorikian/html-proofer) on output to verify there are no missing assets or script files

## Primary Reviewers:
- @lboyette-okta

### CODE REVIEW

File: _source/_docs/api/resources/factors.md [hunk 1]
```diff
@@ -763,7 +763,7 @@ curl -v -X POST \
 
 Customize (and optionally localize) the SMS message sent to the user in case Okta needs to resend the message as part of enrollment.
 
->For instructions about how to create custome templates, see [SMS Template](templates.html#sms-template).
+>For instructions about how to create custome templates, see [SMS Template](/docs/api/resources/templates.html#add-sms-template).
```
Comment (lboyette-okta) [inline _source/_docs/api/resources/factors.md]: nit (saw you fixed the other one): custome => custom

File: scripts/travis.sh [hunk 2]
```diff
@@ -33,3 +33,6 @@ if ! url_consistency_check || ! duplicate_slug_in_url; then
   echo "FAILED LINT CHECK!"
   exit 1;
 fi
+
+# 5. Run htmlproofer to validate links, scripts, and images
+bundle exec htmlproofer ./dist --assume-extension --disable-external --allow-hash-href --empty-alt-ignore --log-level verbose --file-ignore "/3rd_party_notices/","/java_api_sdk/","/python_api_sdk/","/javadoc/"
```
Comment (lboyette-okta) [inline scripts/travis.sh]: maybe add `"/csharp_api_sdk/"`

File: _source/_assets/css/okta/global/_fontawesome.scss [hunk 3]
```diff
@@ -5,12 +5,12 @@
 
 @font-face {
 	font-family: 'FontAwesome';
-	src: url('#{$asset-path}/fonts/fontawesome-webfont.eot?v=4.3.0');
-	src: url('#{$asset-path}/fonts/fontawesome-webfont.eot?#iefix&v=4.3.0') format('embedded-opentype'),
-	url('#{$asset-path}/fonts/fontawesome-webfont.woff2?v=4.3.0') format('woff2'),
-	url('#{$asset-path}/fonts/fontawesome-webfont.woff?v=4.3.0') format('woff'),
-	url('#{$asset-path}/fonts/fontawesome-webfont.ttf?v=4.3.0') format('truetype'),
-	url('#{$asset-path}/fonts/fontawesome-webfont.svg?v=4.3.0#fontawesomeregular') format('svg');
+	src: font_url('fontawesome-webfont.eot?v=4.3.0');
```
Comment (lboyette-okta) [inline _source/_assets/css/okta/global/_fontawesome.scss]: (from the jekyll-assets gem)

File: _source/_assets/css/okta/components/_Header.scss [hunk 4]
```diff
@@ -62,25 +62,25 @@
...
 							border-top: 1px solid get-color('gray-light');
 							margin-top: get-spacing('x-small');
 							padding-top: get-spacing('x-small');
 						}
 					}
-					
+
 					.stack-overflow {
-						background-image: url(#{$asset-path}/img/rounded-stackoverflow--small-blue.png);
+						background-image: image_url("rounded-stackoverflow--small-blue.png");
```
Comment (lboyette-okta) [inline _source/_assets/css/okta/components/_Header.scss]: (also from the jekyll-assets gem)

File: _config.yml [hunk 5]
```diff
@@ -11,9 +11,7 @@ github_username:  oktadeveloper
 gems:
   - jekyll-redirect-from
   - jekyll-paginate
-
-# -- standard exclusions
-exclude: [Gemfile, Gemfile.lock, README.md]
```
Comment (rchild-okta) [inline _config.yml]: They weren't actually included originally - jekyll will read from _source, where these files don't exist. No reason for this to be here originally.

File: _config.yml [hunk 6]
```diff
@@ -41,9 +39,9 @@ production_url: "http://developer.okta.com"
 
 include: [_static]
 
-sass:
-    sass_dir: _sass
```
Comment (rchild-okta) [inline _config.yml]: It's handled by jekyll-assets (they automatically assume scss+css -> css and even do liquid templating in the files w/out needing that front matter stuff).

File: _source/_assets/js/master.js [hunk 7]
```diff
@@ -1,3 +1,5 @@
+//= require vendor/jquery-3.1.0.min
```
Comment (rchild-okta) [inline _source/_assets/js/master.js]: It is their metadata to say "create a bundle with this master.js file and jquery"

File: _source/_code/java/spring_security_saml.md [hunk 8]
```diff
@@ -89,10 +89,7 @@ How to install the Spring Security SAML sample Okta application on Mac OS X:
 
 	This task compiles, tests, and assembles the code into a `.war` file.
 
-	A succesful build should look something like this:
-
-	![succesful build](/assets/img/spring-security-saml-build.png)
-
+	A succesful build should look something like this: {% img spring-security-saml-build.png alt:"successful build" %}
```
Comment (rchild-okta) [inline _source/_code/java/spring_security_saml.md]: Will update these small things

File: _source/_includes/blog_post.html [hunk 9]
```diff
@@ -1,35 +1,35 @@
...
-</div>
+  <div class="wrap">
+    <article class="post-block">
+      <header class="post-title-block">
+        <h1><a href="{{ post.url }}">{{ post.title }}</a></h1>
+        <div class="attribution">
+          {% for post_author in post.author %}
+            {% assign author = site.data.authors[post_author] %}
+            {% if forloop.first %}
+              {% img '{{ author.avatar}}' alt:'{{ author.avatar}}' class:author-avatar %}
```
Comment (rchild-okta) [inline _source/_includes/blog_post.html]: yep, if it's one word

File: _source/_standards/SAML/saml_tracer.md [hunk 10]
```diff
@@ -44,7 +44,7 @@ To create a SAML request for an IdP-initiated flow and inspect it in SAML tracer
 
 You can also start an IdP flow by selecting the App Embed link in a browser (**SAML App** > **General** > **App Embed Link**).
 
-![SAML Response IdP](/assets/img/idp-flow-saml-trace.png "sample idp response in tracer")
+{% img idp-flow-saml-trace.png "sample idp response in tracer" alt:"SAML Response IdP" %}
```
Comment (rchild-okta) [inline _source/_standards/SAML/saml_tracer.md]: oh hmm, good point; I'll delete the other one that's not part of alt

**Pull request merged**